src/algokit_utils/asset.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from typing import TYPE_CHECKING
+
+from algosdk.atomic_transaction_composer import AtomicTransactionComposer, TransactionWithSigner
+from algosdk.transaction import AssetTransferTxn
+
+if TYPE_CHECKING:
+    from algosdk.v2client.algod import AlgodClient
+
+from algokit_utils.models import Account
+
+__all__ = ["opt_in", "opt_out"]
+logger = logging.getLogger(__name__)
+MAX_GROUP_SIZE = 16
+
+
+def ensure_asset_balance(algod_client: "AlgodClient", account: Account, asset_ids: list) -> None:
+    invalid_asset_ids = []
+    for asset_id in asset_ids:
+        try:
+            account_asset_info = algod_client.account_asset_info(account.address, asset_id)
+            if account_asset_info["amount"] != 0:
+                logger.debug(f"Asset {asset_id} balance is not zero")
+                invalid_asset_ids.append(asset_id)
+        except Exception:
+            logger.debug(f"Account ${account.address} does not have asset ${asset_id}")
+            invalid_asset_ids.append(asset_id)
+
+    if not invalid_asset_ids:
+        raise ValueError(
+            f" Assets {invalid_asset_ids} cannot be opted out. Ensure that they are valid and that the "
+            f"account has previously opted into them."
+        )
+
+
+def ensure_asset_first_optin(algod_client: "AlgodClient", account: Account, asset_ids: list) -> None:
```

Comment:
are these exported to external use or only used internally by opt in and opt out? If not exported please make it private